src/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -260,14 +261,32 @@ private static bool TryResolveConflicts<TItem>(this GraphNode<TItem> root, List<
                     if (acceptedLibraries.TryGetValue(childNode.Key.Name, out acceptedNode) &&
                         childNode != acceptedNode &&
                         childNode.Key.VersionRange != null &&
-                        string.Equals(childNode.Key.TypeConstraint, acceptedNode.Key.TypeConstraint, StringComparison.Ordinal) &&
-                        !childNode.Key.VersionRange.Satisfies(acceptedNode.Item.Key.Version))
+                        string.Equals(
+                            childNode.Key.TypeConstraint, 
+                            acceptedNode.Key.TypeConstraint, 
+                            StringComparison.Ordinal))
                     {
-                        versionConflicts.Add(new VersionConflictResult<TItem>
+                        var versionRange = childNode.Key.VersionRange;
+                        var checkVersion = acceptedNode.Item.Key.Version;
+
+                        // Allow prerelease versions if the selected library is prerelease and the range is
+                        // using the default behavior of filtering to stable versions.
+                        // Ex: [4.0.0, ) should allow 4.0.10-beta if that library was selected during the graph walk
+                        // The decision on if a prerelease version should be allowed should happen previous to this
+                        // check during the walk.
+                        if (checkVersion.IsPrerelease && !versionRange.IncludePrerelease)
```

Comment:
@deepakaravindr this code could just always set it to include prerelease and it would work since the filtering is done before hand. But what this if statement is really doing here is checking if the resolved package is prerelease before updating the range to allow prerelease. In SetIncludePrelease it does the check just to noop if it can, which is different.